execution/telemetry/telemetry.go

```diff
@@ -203,8 +203,18 @@ func (t *Operator) Series(ctx context.Context) ([]labels.Labels, error) {
 
 func (t *Operator) Next(ctx context.Context) ([]model.StepVector, error) {
 	start := time.Now()
+	totalSamplesBefore := t.OperatorTelemetry.Samples().TotalSamples
+
 	defer func() { t.OperatorTelemetry.AddNextExecutionTime(time.Since(start)) }()
-	return t.inner.Next(ctx)
+	out, err := t.inner.Next(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	totalSamplesAfter := t.OperatorTelemetry.Samples().TotalSamples
+	t.OperatorTelemetry.Samples().UpdatePeak(int(totalSamplesAfter) - int(totalSamplesBefore))
```

Comment:
nit: Should this be just t.OperatorTelemetry.UpdatePeak()

Reply:
Also, we no longer need the other calls to `UpdatePeak`. Could you remove it from `IncrementSamplesAtTimestamp()`.

Reply:
I think we should add a new method in the interface to UpdatePeak()

```
type OperatorTelemetry interface {
	fmt.Stringer

	MaxSeriesCount() int
	SetMaxSeriesCount(count int)
	ExecutionTimeTaken() time.Duration
	AddSeriesExecutionTime(time.Duration)
	SeriesExecutionTime() time.Duration
	AddNextExecutionTime(time.Duration)
	NextExecutionTime() time.Duration
	IncrementSamplesAtTimestamp(samples int, t int64)
	UpdatePeak(samples int)
	Samples() *stats.QuerySamples
	LogicalNode() logicalplan.Node
}
```